### DISCUSSION TITLE
umi4打包的拆分文件全在head里，怎么像umi3那样控制放到body里呢？

### DISCUSSION THREAD

**Discussion by Vantown** [umijs/umi#11189] [Q&A]
![image](https://github.com/umijs/umi/assets/29136022/f99bc80c-d80a-47f6-ad0a-71ce8d259d94)

**Comment by fz6m** [comment]
由于 react 会在所有代码加载完毕后再运行，所以在 head 里是不影响应用的。

**Reply by Vantown** [reply]
> 由于 react 会在所有代码加载完毕后再运行，所以在 head 里是不影响应用的。

现在我们的应用做的有点大，我把资源都放到body里在umi那附近加载，body开始的地方放个加载动画，这样不会有白屏等待的时间，我们的应用打完包将近10M，等加载完时间有点长了，所以挪到body里加载好一些

**Reply by Vantown** [reply]
> 由于 react 会在所有代码加载完毕后再运行，所以在 head 里是不影响应用的。

有没有什么方式可以控制最终打包出来的资源在html出现的位置呢？比如说把打出来的文件可以通过特定api挪到umi的后面或者前面，modifyHTML能达到这种效果吗？

**Reply by fz6m** [reply]
可以自己尝试一下，不推荐这么做，如果出现了问题需要自己承担。

**Reply by Vantown** [reply]
> 可以自己尝试一下，不推荐这么做，如果出现了问题需要自己承担。

好的知道了，多谢

**Comment by xiaoxuehai** [comment]
兄弟  这个问题你解决了吗
